### DISCUSSION TITLE
How to enable backends in GET /me ?

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#996] [Q&A] [answered]
I don't quite understand how to inject a dependency for the current user in the `GET /me` route of the Users router, so that it accepts a `get_enabled_backends` parameter. I have the feeling that we should redefine this route, but I'm not sure.

**Comment by frankie567** [comment, marked as answer]
Yes, you would need to redefine the route yourself if you need to pass a custom `current_user` dependency. By default, it uses a simple `current_user(active=True)` one.

**Reply by davidbrochart** [reply]
Thanks, do you recommend a better way to redefine a route than doing something like this?
```py
async def get_me(user: UserRead = Depends(current_user)):
    return user

users_router = fapi_users.get_users_router(UserRead, UserUpdate)
for route in users_router.routes:
    if route.path == "/me":
        break
route.endpoint = get_me
```

**Reply by frankie567** [reply]
Maybe better to just define it as a regular route before the router (first matching route wins in FastAPI):

```py
router = APIRouter()

@router.get("/me")
async def get_me(user: UserRead = Depends(current_user)):
    return user

router.add_router(fastapi_users.get_users_router(UserRead, UserUpdate))
```

**Reply by davidbrochart** [reply]
We cannot do that because [FPS](https://github.com/jupyter-server/fps) forbids router duplication, so I'll stick with my inelegant solution :smile: 
Thanks!

**Reply by frankie567** [reply]
Oh I see! That said, from what I understand of [this code](https://github.com/jupyter-server/fps/blob/df59b19eaf530efbd18ea9e30c78bdd53d94d629/fps/app.py#L217-L229), it doesn't check for path duplication *inside* a single router, right? (might no be an expected behavior though 😏)

**Reply by davidbrochart** [reply]
Indeed, but it looks like `add_router` doesn't exist, there is `add_route` and `include_router`. I've tried with `include_router` but it doesn't seem to work.

**Reply by frankie567** [reply]
Yes, sorry, I meant `include_router` 😅 It's weird that it doesn't work, I've tried on a simple example on my side and the first router gets called.

**Reply by davidbrochart** [reply]
My bad, I added a `prefix="/auth/user"` to the `get_me` route that already had `"/auth/user/me"` :smile: 
Works like a charm, thanks!

**Reply by frankie567** [reply]
Always obvious after a good sleep night 😄
